sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthFileCredential.java

```diff
@@ -0,0 +1,120 @@
+ * client secret, client certificates are not supported at this time.
+ */
+@Immutable
+public class AuthFileCredential implements TokenCredential {
+    /* The file path value. */
+    private final String filepath;
+    private  IdentityClientOptions identityClientOptions;
+    private TokenCredential credential;
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    /**
+     *  Creates an instance of the SdkAuthFileCredential class based on information in given SDK Auth file.
+     *  If the file is not found or there are errors parsing it, <see cref="GetToken(TokenRequestContext)"/>
+     *  and <see cref="GetToken(TokenRequestContext)"/> will throw a <see cref="Exception"/>
+     *  with details on why the file could not be used.
+     *
+     * @param filepath The path to the SDK Auth file.
+     * @param identityClientOptions the options for configuring the identity client
+     */
+    AuthFileCredential(String filepath, IdentityClientOptions identityClientOptions) {
+        Objects.requireNonNull(filepath, "'filepath' cannot be null.");
+        Objects.requireNonNull(identityClientOptions, "'identityClientOptions' cannot be null.");
+        this.filepath = filepath;
+        this.identityClientOptions=identityClientOptions;
+    }
+
+    /**
+     * Obtains a token from the Azure Active Directory service, using the specified client detailed specified in the SDK Auth file.
+     * This method is called by Azure SDK clients. It isn't intended for use in application code.  
+     * If the SDK Auth file is missing or invalid, this method throws a <see cref="AuthenticationFailedException"/> exception.
+     * <param name="requestContext">The details of the authentication request</param>
+     * <returns>An <see cref="AccessToken"/> which can be used to authenticate service client calls.</returns>
+     */
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        try {
+            ensureCredential();
+        } catch (Exception e) {
+            return Mono.error(e);
+        }
+        return credential.getToken(request);
+    }
+    
+    /**  
+     * Ensures that credential information is loaded from the SDK Auth file. This method should be called to initialize
+     * <code>_credential</code> before it is used. If the SDK Auth file is not found or invalid, this method will throw
+     * <see cref="AuthenticationFailedException"/>.
+     * <returns>A method that will ensure <code>credential</code> has been initialized</returns>
+     */
+    private void ensureCredential() throws Exception {
+        if(credential ==null){
+            try
+            {
+                credential = BuildCredentialForCredentialsFile(ParseCredentialsFile(filepath));
+            } catch (Exception e){  
+                throw new Exception("Error parsing SDK Auth File", e);
```

Comment:
Good to add logger.logExceptionAsError(new Exception....)
logger can be used from azure core
Ideally, would like to use a better Exception class than just Exception, like InvalidArgumentException.

Plus, IOException will need to be handled separately, and should be bubbled up as it is.